libraries/AP_Compass/AP_Compass.h

```diff
@@ -183,7 +183,7 @@ friend class AP_Compass_Backend;
 
     /// Return the health of a compass
     bool healthy(uint8_t i) const { return _get_state(Priority(i)).healthy; }
-    bool healthy(void) const { return healthy(0); }
```

Comment:
Is this function used in contexts that check whether the primary compass is healthy? If so, would it make sense to avoid touching that logic and write an `all_healthy()` function instead?

Reply:
Actually, looking at the code again now I don't think this change is needed because the healthy mask that gets reported in the named value int you added doesn't use this, right?

Reply:
This function "compass.healthy()" is used in arming check and in the system status sensor_health. If any one compass is down it will be 0/false.

Reply:
Here are all the places that uses "compass.healthy()"

https://github.com/matternet/ardupilot/blob/mttr-rebase-4.0/ArduCopter/AP_Arming.cpp#L620
https://github.com/matternet/ardupilot/blob/mttr-rebase-4.0/ArduCopter/Log.cpp#L268
https://github.com/matternet/ardupilot/blob/mttr-rebase-4.0/libraries/AP_Arming/AP_Arming.cpp#L394
https://github.com/matternet/ardupilot/blob/mttr-rebase-4.0/libraries/GCS_MAVLink/GCS.cpp#L137

Reply:
@dat-tran-matternet You're right -- we require all compasses to be healthy in each of those cases. Thank you for clarifying 👍